src/init.cpp

```diff
@@ -588,7 +588,11 @@ void SetupServerArgs(ArgsManager& argsman)
     argsman.AddArg("-dustrelayfee=<amt>", strprintf("Fee rate (in %s/kvB) used to define dust, the value of an output such that it will cost more than its value in fees at this fee rate to spend it. (default: %s)", CURRENCY_UNIT, FormatMoney(DUST_RELAY_TX_FEE)), ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::NODE_RELAY);
     argsman.AddArg("-bytespersigop", strprintf("Equivalent bytes per sigop in transactions for relay and mining (default: %u)", DEFAULT_BYTES_PER_SIGOP), ArgsManager::ALLOW_ANY, OptionsCategory::NODE_RELAY);
     argsman.AddArg("-datacarrier", strprintf("Relay and mine data carrier transactions (default: %u)", DEFAULT_ACCEPT_DATACARRIER), ArgsManager::ALLOW_ANY, OptionsCategory::NODE_RELAY);
-    argsman.AddArg("-datacarriersize", strprintf("Maximum size of data in data carrier transactions we relay and mine (default: %u)", MAX_OP_RETURN_RELAY), ArgsManager::ALLOW_ANY, OptionsCategory::NODE_RELAY);
+    argsman.AddArg("-datacarriersize",
+                   strprintf("Relay and mine transactions whose data-carrying raw scriptPubKey "
+                             "is of this size or less (default: %u)",
+                             MAX_OP_RETURN_RELAY),
+                   ArgsManager::ALLOW_ANY, OptionsCategory::NODE_RELAY);
```

Comment:
NACK This wording is misleading. OP_Return outputs only contain data if they contain something in addition to the OP_Return opcode.

A better approach here if we are touching this opcode could be to redefine `-datacarriersize` to be size *in-addition-too* the OP_Return opcode that is allowed. Thus `-datacarriersize=0` would allow 0 bytes in addition to the 1 byte used by the OP_Return opcode.

Reply:
Since you are updating this, you should change the doc text to make clear that the current behavior is a mistake that needs fixing rathre than continue to incorrectly call them data carrier scriptPubKey's.

But it'd be less code churn to just fix the behavior in one shot. We have an ACK on https://github.com/bitcoin/bitcoin/pull/27261, so there is an argument to fix the bare OP_Return case. Rather than merging https://github.com/bitcoin/bitcoin/pull/27261 as-is I could make a pull-req that simply fixes the `-datacarriersize` off-by-one error. Fixing that error would also allow `-datacarrier` to be dropped entirely, as it is redundant: `-datacarriersize=0` prohibits data carrying outputs just fine.